activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/model/Extension.java

```diff
@@ -0,0 +1,29 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.spring.process.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Extension {
+
+    private Map<String, VariableDefinition> properties = new HashMap<>();
```

Comment:
@ryandawsonuk let's rename this to variables here.. and use Jackson annotations to map to the "properties" field in the json.. this is what Constantin is doing in modeling as well.